src/main/java/net/minecraftforge/fml/relauncher/libraries/LibraryManager.java

```diff
@@ -0,0 +1,466 @@
+                    {
+                        ZipEntry ze = null;
+                        while ((ze = zi.getNextEntry()) != null)
+                        {
+                            if (ze.getName().equalsIgnoreCase(JarFile.MANIFEST_NAME))
+                            {
+                                manifest_data = readAll(zi);
+                                meta = new Manifest(new ByteArrayInputStream(manifest_data)).getMainAttributes();
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (meta == null || !meta.containsKey(MAVEN_ARTIFACT)) //Ugh I really don't want to do backwards compatibility here, I want to force modders to provide information... TODO: Remove in 1.13?
+                {
+                    boolean found = false;
+                    for (File dir : modDirs)
+                    {
+                        File target = new File(dir, depEndName);
+                        if (target.exists())
+                        {
+                            FMLLog.log.debug("Found existing ContainDep extracted to {}, skipping extraction", target.getCanonicalPath());
+                            found = true;
+                        }
+                    }
+                    if (!found)
+                    {
+                        File target = new File(modDirs[0], depEndName);
+                        FMLLog.log.debug("Extracting ContainedDep {} from {} to {}", dep, jar.getName(), target.getCanonicalPath());
+                        try
+                        {
+                            Files.createParentDirs(target);
+                            try
+                            (
+                                FileOutputStream out = new FileOutputStream(target);
+                                InputStream in = data == null ? jar.getInputStream(depEntry) : new ByteArrayInputStream(data)
+                            )
+                            {
+                                ByteStreams.copy(in, out);
+                            }
+                            FMLLog.log.debug("Extracted ContainedDep {} from {} to {}", dep, jar.getName(), target.getCanonicalPath());
+                            extractPacked(target, modlist, modDirs);
+                        }
+                        catch (IOException e)
+                        {
+                            FMLLog.log.error("An error occurred extracting dependency", e);
+                        }
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        String timestamp = meta.getValue(TIMESTAMP);
+                        if (timestamp != null)
+                            timestamp = SnapshotJson.TIMESTAMP.format(new Date(Integer.parseInt(timestamp)));
```

Comment:
This limits the timestamp to Jan 25 21:31:23 1970, as this Date constructor takes a long in milliseconds. Did you mean `Long.parseLong(timestamp)`?